okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -465,98 +464,104 @@ public void run() {
             maxConcurrentStreams = Integer.MAX_VALUE;
             startPendingStreams();
           }
-          frameWriter.becomeConnected(testFrameWriter, socket);
           connectedFuture.set(null);
-          return;
         }
+      });
+      return null;
+    }
 
-        // Use closed source on failure so that the reader immediately shuts down.
-        BufferedSource source = Okio.buffer(new Source() {
-          @Override
-          public long read(Buffer sink, long byteCount) {
-            return -1;
-          }
+    final AsyncSink asyncSink = AsyncSink.sink(serializingExecutor, this);
+    final Variant variant = new Http2();
+    final FrameWriter rawFrameWriter = variant.newWriter(Okio.buffer(asyncSink), true);
 
-          @Override
-          public Timeout timeout() {
-            return Timeout.NONE;
-          }
+    synchronized (lock) {
+      frameWriter = new ExceptionHandlingFrameWriter(this, rawFrameWriter);
+      outboundFlow = new OutboundFlowController(this, frameWriter, initialWindowSize);
 
-          @Override
-          public void close() {}
-        });
-        Variant variant = new Http2();
-        BufferedSink sink;
-        Socket sock;
-        SSLSession sslSession = null;
-        try {
-          if (proxy == null) {
-            sock = new Socket(address.getAddress(), address.getPort());
-          } else {
-            sock = createHttpProxySocket(
-                address, proxy.proxyAddress, proxy.username, proxy.password);
-          }
+      // Connecting in the serializingExecutor, so that some stream operations like synStream
+      // will be executed after connected.
+      serializingExecutor.execute(new Runnable() {
```

Comment:
Including this execute within the lock seems unnecessary and indents the code making it very difficult to review. Could it be removed from the lock to be unindented?